include/forcing/NullForcingProvider.hpp

```diff
@@ -0,0 +1,271 @@
+     * @return Whether the property's value is an aggregate sum.
+     */
+    //TODO this one used in Bmi_Module_Formulation.hpp and Bmi_Multi_Formulation.hpp
+    inline bool is_property_sum_over_time_step(const std::string& name) override {
+        return is_param_sum_over_time_step(name);
+    }
+
+    const std::vector<std::string> &get_available_variable_names() override {
+        return available_forcings;
+    }
+
+    private:
+
+    /**
+     * @brief Checks forcing vector index bounds and adjusts index if out of vector bounds
+     * /// \todo: Bounds checking is based on precipitation vector. Consider potential for vectors of different sizes and indices.
+     */
+    inline void check_forcing_vector_index_bounds()
+    {
+        //Check if forcing index is less than zero and if so, set to zero.
+        if (forcing_vector_index < 0)
+        {
+            forcing_vector_index = 0;
+            /// \todo: Return appropriate warning
+            std::cout << "WARNING: Forcing vector index is less than zero. Therefore, setting index to zero." << std::endl;
+        }
+
+        //Check if forcing index is greater than or equal to the size of the size of the time vector and if so, set to zero.
+        else if (forcing_vector_index >= time_epoch_vector.size())
+        {
+            forcing_vector_index = time_epoch_vector.size() - 1;
+            /// \todo: Return appropriate warning
+            std::cout << "WARNING: Reached beyond the size of the forcing vector. Therefore, setting index to last value of the vector." << std::endl;
+        }
+
+        return;
+    }
+
+    /**
+     * Get the current value of a forcing param identified by its name.
+     *
+     * @param name The name of the forcing param for which the current value is desired.
+     * @param index The index of the desired forcing time step from which to obtain the value.
+     * @return The particular param's value at the given forcing time step.
+     */
+    inline double get_value_for_param_name(const std::string& name, int index) {
+        if (index >= time_epoch_vector.size() ) {
+            throw std::out_of_range("Forcing had bad index " + std::to_string(index) + " for value lookup of " + name);
+        }
+
+        std::string can_name = name;
+        if(data_access::WellKnownFields.count(can_name) > 0){
+            auto t = data_access::WellKnownFields.find(can_name)->second;
+            can_name = std::get<0>(t);
+        }
+
+        if (forcing_vectors.count(can_name) > 0) {
+            return forcing_vectors[can_name].at(index);
+        }
+        else {
+            throw std::runtime_error("Cannot get forcing value for unrecognized parameter name '" + name + "'.");
+        }
+    }
+
+    /**
+     * @brief Read Forcing Data from CSV
+     * Reads only data within the specified model start and end date-times.
+     * @param file_name Forcing file name
+     */
+    // No file to read
+    void read_csv()
```

Comment:
I tried to remove it entirely. but that cuased a problem. Somehow we need at least one available forcing variable names to the Formulation_Manager to retrieve the start and stop time. Otherwise, it causes a runtime error.

Reply:
Interesting. So, if this is the case, then we might need to move to the `ZerosDataProvider` as discussed in #569. However, I still wonder if this can be avoided. I think I know where that is coming from, let me try to find it.